### ISSUE TITLE
Answer blank regression: first submit click ignored with multiple blanks

### ISSUE THREAD

**Issue by dqnykamp** [Doenet/DoenetTools#1215]
When an answer has multiple input blanks, and a mathinput or textinput is focused, clicking the submit button is ignored.

Mathinput example
```
Enter two values that sum to 1
<answer>
  <mathinput name="num2" />
  <mathinput name="num1" />
  <award>
    <when>
      $num1 + $num2 = 1
    </when>
  </award>
</answer>
```

Textinput example
```
Enter two articles:
<answer>
  <textinput name="a1" />
  <textinput name="a2" />
  <award><when>
    ($a1 = a or $a1 = an or $a1 = the) 
    and 
    ($a2 = a or $a2 = an or $a2 = the)
    and
    $a1 != $a2
  </when></award>
</answer>
```

Mixed mathinput and textinput example
```
Enter 3 birds:
<answer>
  <mathinput name="n" />
  <textinput name="birds" />
  <award>
    <when>
      $n = 3 and $birds=birds
    </when>
  </award>
</answer>
```

With a mix of mathinput and choiceinput, it works properly if choiceinput is focused, but not if mathinput is focused.

```
Enter 3 birds:
<answer>
  <mathinput name="n" />
  <choiceinput name="birds" inline >
    <choice credit="1">birds</choice>
    <choice>mammals</choice>
  </choiceinput>
  <award>
    <when>
      $n = 3 and $birds=birds
    </when>
  </award>
</answer>
```

**Comment by clary149** [comment]
Keagan and I have both worked on this issue. For the one input field in answer tag, I was able to add an if statement in the onBlur that checks if a checkWork button was the related target and then calls submitAnswer when the id of the checkWork button associated with that input field matches the id of the related target. When there are 2 input fields, I cannot find the id of the checkWork button that goes with them, so this if statement does not fix the problem. When we checkout the commit immediately before Keagan's styling fixes for answer tag and mathInput (#1174), these bugs do not exist. Both of us have reviewed this pr and cannot find anything that would change the functionality of these components. Another thing to note is that textInput and mathInput both hold focus, so that multiple buttons on the keyboard can be clicked without losing the input field. ChoiceInput does not have this bug and does not have focus modified for keyboard. We don't know what to try next and would appreciate any further ideas!

**Comment by dqnykamp** [comment]
I'm wondering if style-components is causing this problem.   Will creating the StyleComponents outside the function component and resolving this styled-components warning help?

> styled-components.v5.3.1.js:236 The component styled.button with the id of "sc-jSgvzq" has been created dynamically.
> You may see this warning because you've called styled inside another component.
> To resolve this only create new StyledComponents outside of any render method and function component.
